syntax = "proto3";
package core.protos.connectors;

import "core/protos/base.proto";
import "google/protobuf/wrappers.proto";
import "google/protobuf/struct.proto";
import "core/protos/connectors/connector.proto";
import "core/protos/assets/cloudwatch_asset.proto";
import "core/protos/assets/grafana_asset.proto";
import "core/protos/assets/clickhouse_asset.proto";
import "core/protos/assets/slack_asset.proto";
import "core/protos/assets/newrelic_asset.proto";
import "core/protos/assets/datadog_asset.proto";
import "core/protos/assets/postgres_asset.proto";
import "core/protos/assets/eks_asset.proto";
import "core/protos/assets/azure_asset.proto";
import "core/protos/assets/bash_asset.proto";
import "core/protos/assets/gke_asset.proto";
import "core/protos/assets/elastic_search_asset.proto";
import "core/protos/assets/gcm_asset.proto";
import "core/protos/assets/open_search_asset.proto";
import "core/protos/assets/asana_asset.proto";
import "core/protos/assets/github_asset.proto";
import "core/protos/assets/jira_asset.proto";
import "core/protos/assets/argocd_asset.proto";
import "core/protos/assets/jenkins_asset.proto";
import "core/protos/assets/signoz_asset.proto";
import "core/protos/assets/posthog_asset.proto";
import "core/protos/assets/sql_database_asset.proto";
import "core/protos/assets/mongodb_asset.proto";

message ConnectorModelTypeOptions {
  ConnectorMetadataModelType model_type = 1;
  oneof options {
    CloudwatchLogGroupAssetOptions cloudwatch_log_group_model_options = 2;
    CloudwatchMetricAssetOptions cloudwatch_metric_model_options = 3;
    GrafanaTargetMetricPromQlAssetOptions grafana_target_metric_promql_model_options = 4;
    ClickhouseDatabaseAssetOptions clickhouse_database_model_options = 5;
    SlackChannelAssetOptions slack_channel_model_options = 6;
    NewRelicApplicationEntityAssetOptions new_relic_entity_application_model_options = 7;
    NewRelicDashboardEntityAssetOptions new_relic_entity_dashboard_model_options = 8;
    DatadogServiceAssetOptions datadog_service_model_options = 9;
    PostgresDatabaseAssetOptions postgres_database_model_options = 10;
    EksClusterAssetOptions eks_cluster_model_options = 11;
    BashSshServerAssetOptions ssh_server_model_options = 12;
    AzureWorkspaceAssetOptions azure_workspace_model_options = 13;
    GrafanaDatasourceAssetOptions grafana_prometheus_datasource_model_options = 14;
    GkeClusterAssetOptions gke_cluster_model_options = 15;
    ElasticSearchIndexAssetOptions elastic_search_index_model_options = 16;
    OpenSearchIndexAssetOptions open_search_index_model_options = 17;
    GcmMetricAssetOptions gcm_metric_model_options = 18;
    DatadogServiceAssetOptions datadog_oauth_service_model_options = 19;
    CloudwatchLogGroupQueryAssetOptions cloudwatch_log_group_query_model_options = 20;
    AsanaProjectAssetOptions asana_project_model_options = 21;
    GrafanaAlertRuleAssetOptions grafana_alert_rule_model_options = 22;
    CloudwatchAlarmAssetOptions cloudwatch_alarm_model_options = 23;
    DatadogDashboardAssetOptions datadog_dashboard_model_options = 24;
    DatadogDashboardAssetOptions datadog_oauth_dashboard_model_options = 25;
    GithubRepositoryAssetOptions github_repository_model_options = 26;
    GithubMemberAssetOptions github_member_model_options = 27;
    JiraUserAssetOptions jira_user_model_options = 28;
    JiraProjectAssetOptions jira_project_model_options = 29;
    ArgoCDAppsAssetOptions argocd_apps_model_options = 30;
    AzureResourceAssetOptions azure_resource_model_options = 31;
    JenkinsAppsAssetOptions jenkins_apps_model_options = 32;
    MongoDBDatabaseAssetOptions mongodb_database_model_options = 33;
    MongoDBCollectionAssetOptions mongodb_collection_model_options = 34;
    PosthogPropertyAssetOptions posthog_property_model_options = 35;
    EcsClusterAssetOptions ecs_cluster_model_options = 36;
    EcsTaskAssetOptions ecs_task_model_options = 37;
    EcsServiceAssetOptions ecs_service_model_options = 38;
    GcmDashboardEntityAssetOptions gcm_dashboard_model_options = 39;
    GcmCloudRunServiceAssetOptions gcm_cloud_run_service_model_options = 40;
    SignozDashboardAssetOptions signoz_dashboard_model_options = 41;
    SignozAlertAssetOptions signoz_alert_model_options = 42;
  }
}

message AccountConnectorAssets {
  Connector connector = 1;
  oneof assets {
    CloudwatchAssets cloudwatch = 101;
    GrafanaAssets grafana = 102;
    ClickhouseAssets clickhouse = 103;
    SlackAssets slack = 104;
    NewRelicAssets new_relic = 105;
    DatadogAssets datadog = 106;
    PostgresAssets postgres = 107;
    EksAssets eks = 108;
    BashAssets bash = 109;
    AzureAssets azure = 110;
    GkeAssets gke = 111;
    ElasticSearchAssets elastic_search = 112;
    GcmAssets gcm = 113;
    DatadogAssets datadog_oauth = 15;
    OpenSearchAssets open_search = 16;
    AsanaAssets asana = 17;
    GithubAssets github = 18;
    JiraAssets jira_cloud = 19;
    ArgoCDAssets argocd = 20;
    JenkinsAssets jenkins = 21;
    MongoDBAssets mongodb = 22;
    PosthogAssets posthog = 23;
    SqlAssets sql = 24;
    SignozAssets signoz = 25;
  }
}

message AccountConnectorAssetsModelFilters {
  oneof filters {
    CloudwatchLogGroupAssetOptions cloudwatch_log_group_model_filters = 1;
    CloudwatchMetricAssetOptions cloudwatch_metric_model_filters = 2;
    GrafanaTargetMetricPromQlAssetOptions grafana_target_metric_promql_model_filters = 3;
    ClickhouseDatabaseAssetOptions clickhouse_database_model_filters = 4;
    SlackChannelAssetOptions slack_channel_model_filters = 5;
    NewRelicApplicationEntityAssetOptions new_relic_entity_application_model_filters = 6;
    NewRelicDashboardEntityAssetOptions new_relic_entity_dashboard_model_filters = 7;
    DatadogServiceAssetOptions datadog_service_model_filters = 8;
    PostgresDatabaseAssetOptions postgres_database_model_filters = 9;
    EksClusterAssetOptions eks_cluster_model_filters = 10;
    BashSshServerAssetOptions ssh_server_model_filters = 11;
    AzureWorkspaceAssetOptions azure_workspace_model_filters = 12;
    GrafanaDatasourceAssetOptions grafana_prometheus_datasource_model_filters = 13;
    GkeClusterAssetOptions gke_cluster_model_filters = 14;
    ElasticSearchIndexAssetOptions elastic_search_index_model_filters = 15;
    OpenSearchIndexAssetOptions open_search_index_model_filters = 16;
    GcmMetricAssetOptions gcm_metric_model_filters = 17;
    DatadogServiceAssetOptions datadog_oauth_service_model_filters = 18;
    CloudwatchLogGroupQueryAssetOptions cloudwatch_log_group_query_model_filters = 19;
    AsanaProjectAssetOptions asana_project_model_filters = 20;
    GrafanaAlertRuleAssetOptions grafana_alert_rule_model_filters = 21;
    CloudwatchAlarmAssetOptions cloudwatch_alarm_model_filters = 22;
    DatadogDashboardAssetOptions datadog_dashboard_model_filters = 23;
    DatadogDashboardAssetOptions datadog_oauth_dashboard_model_filters = 24;
    GithubRepositoryAssetOptions github_repository_model_filters = 25;
    GithubMemberAssetOptions github_member_model_filters = 26;
    JiraUserAssetOptions jira_user_model_filters = 27;
    JiraProjectAssetOptions jira_project_model_filters = 28;
    ArgoCDAppsAssetOptions argocd_apps_model_filters = 29;
    AzureResourceAssetOptions azure_resource_model_filters = 30;
    GrafanaDashboardAssetOptions grafana_dashboard_model_filters = 31;
    JenkinsAppsAssetOptions jenkins_apps_model_filters = 32;
    MongoDBDatabaseAssetOptions mongodb_database_model_filters = 33;
    MongoDBCollectionAssetOptions mongodb_collection_model_filters = 34;
    PosthogPropertyAssetOptions posthog_property_model_filters = 35;
    SqlTableAssetOptions sql_table_model_filters = 36;
    EcsClusterAssetOptions ecs_cluster_model_filters = 39;
    EcsTaskAssetOptions ecs_task_model_filters = 37;
    EcsServiceAssetOptions ecs_service_model_filters = 38;
    GcmDashboardEntityAssetOptions gcm_dashboard_model_filters = 40;
    GcmCloudRunServiceAssetOptions gcm_cloud_run_service_model_filters = 41;
    CloudwatchDashboardAssetOptions cloudwatch_dashboard_model_filters = 42;
  }
}